compiler/rustc_data_structures/src/memmap.rs

```diff
@@ -11,33 +12,48 @@ pub struct Mmap(Vec<u8>);
 
 #[cfg(not(any(miri, target_arch = "wasm32")))]
 impl Mmap {
-    /// # Safety
-    ///
     /// The given file must not be mutated (i.e., not written, not truncated, ...) until the mapping is closed.
     ///
-    /// However in practice most callers do not ensure this, so uses of this function are likely unsound.
+    /// This process must not modify nor remove the backing file while the memory map lives.
```

Comment:
Why would removing the file be a problem? Mappings should say valid for unlinked files... unless network filesystems are involved I guess...